### PR TITLE
Add a utility to query the memory usage percentage of a configured redis cluster

### PR DESCRIPTION
**Pull request by Swatinem**
The memory usage of Redis, in particular the processing store, will be used as one of the metrics for backpressure management.

### CODE REVIEW

File: src/sentry/processing/backpressure/redis.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if isinstance(cluster, RedisCluster):
+        # `RedisCluster` returns these as a dictionary, with the node-id as key
+        yield from cluster.info().values()
+    elif isinstance(cluster, rb.Cluster):
+        # for `rb`, we have to iterate over all the clients ourselves
+        with cluster.all() as client:
+            yield client.info()
+    else:
+        # otherwise, lets just hope that `info()` does the right thing
+        yield cluster.info()
```
Comment (asottile-sentry) [inline src/sentry/processing/backpressure/redis.py:60]: these types don't match

File: src/sentry/processing/backpressure/redis.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import rb
+from redis import Redis
+from rediscluster import RedisCluster
+
+# Based on configuration, this could be:
+# - a `rb` (aka redis blaster) Cluster
+# - a `rediscluster` Cluster (actually `RetryingRedisCluster`)
+# - a straight `Redis` client (actually `FailoverRedis`)
+# - or any class configured via `client_class`.
+Cluster = Union[rb.Cluster, RedisCluster, Redis]
```
Comment (asottile-sentry) [inline src/sentry/processing/backpressure/redis.py:12]: taking a union like this feels like a smell -- it should be 3 separate apis for each of the types

Comment (untitaker) [reply src/sentry/processing/backpressure/redis.py:12]: almost all of our redis-based services attempt to be agnostic across at least two of those types. I don't disagree that we could have better abstractions for those, but whether something points to a redis cluster or a single-node redis is runtime-configuration, therefore `sentry.utils.redis.cluster_manager.get` returns a union, and this code just inherits this design decision for the sake of consistency

File: src/sentry/processing/backpressure/redis.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from typing import Any, Generator, Mapping, Sequence, Union
+
+import rb
+from redis import Redis
+from rediscluster import RedisCluster
+
+# Based on configuration, this could be:
+# - a `rb` (aka redis blaster) Cluster
```
Comment (untitaker) [inline src/sentry/processing/backpressure/redis.py:8]: I would only support redis and rediscluster, not rb. if we want to monitor a rb instance, we should take the opportunity and move it to a different setup

File: src/sentry/processing/backpressure/redis.py (hunk 1 above)
Comment (Swatinem) [reply src/sentry/processing/backpressure/redis.py]: They actually do? Or what do you mean?

**Review by untitaker — APPROVED** [review]

**Pull request merged**